chapters/01-spatial-data.qmd

```diff
@@ -21,3 +29,118 @@ using GeoMakie
 
 f, a, p = poly(df.geom)
 ```
+
+
+### Raster from scratch {#sec-raster-from-scratch}
+
+In this section, we are going to demonstrate the creation of rasters from scratch.
+We will construct two small rasters, `elev` and `grain`, which we will use in examples later in the book.
+Unlike creating a vector layer (see @sec-vector-layer-from-scratch), creating a raster from scratch is rarely needed in practice because aligning a raster with the proper spatial extent is challenging to do programmatically ("georeferencing" tools in GIS software are a better fit for the job).
+Nevertheless, the examples will be helpful to become more familiar with the **Rasters.jl** data structures.
+
+```{julia}
+using Rasters
+import GeoFormatTypes as GFT
+```
+
+Conceptually, a raster is an array combined with georeferencing information, whereas the latter comprises:
+
+-   A transformation matrix, containing the origin and resolution, thus linking pixel indices with coordinates in a particular coordinate system
+-   A CRS definition, specifying the association of that coordinate system with the surface of the earth (optional)
+
+Therefore, to create a raster, we first need to have an array with the values, and then supplement it with the georeferencing information.
+Let's create the arrays `elev` and `grain`.
+The `elev` array is a $6 \times 6$ array with sequential values from `1` to `36`.
+It can be created as follows using base Julia functions.
+
+```{julia}
+elev = reshape(UInt8(1):UInt8(36), (6, 6))
+```
+
+The `grain` array represents a categorical raster with values `0`, `1`, `2`, corresponding to categories "clay", "silt", "sand", respectively.
+We will create it from a specific arrangement of pixel values, using  `reshape`.
```

Comment:
Does Rasters support CategoricalArrays? Might be nice to introduce some Julia concepts of combining everything seamlessly?

Reply:
It does but I don't know if that will write to file correctly